volume/local/local_unix.go

```diff
@@ -112,18 +113,34 @@ func (v *localVolume) mount() error {
 	if v.opts.MountDevice == "" {
 		return fmt.Errorf("missing device in volume options")
 	}
+
 	mountOpts := v.opts.MountOpts
+	mountDevice := v.opts.MountDevice
+
 	switch v.opts.MountType {
-	case "nfs", "cifs":
+	case "nfs":
 		if addrValue := getAddress(v.opts.MountOpts); addrValue != "" && net.ParseIP(addrValue).To4() == nil {
 			ipAddr, err := net.ResolveIPAddr("ip", addrValue)
 			if err != nil {
 				return errors.Wrapf(err, "error resolving passed in network volume address")
 			}
 			mountOpts = strings.Replace(mountOpts, "addr="+addrValue, "addr="+ipAddr.String(), 1)
 		}
+	case "cifs":
+		deviceUrl, err := url.Parse(v.opts.MountDevice)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing mount device url")
+		}
+		if deviceUrl.Host != "" && net.ParseIP(deviceUrl.Hostname()) == nil {
+			ipAddr, err := net.ResolveIPAddr("ip", deviceUrl.Hostname())
+			if err != nil {
+				return errors.Wrapf(err, "error resolving passed in network volume address")
+			}
+			deviceUrl.Host = net.JoinHostPort(ipAddr.String(), deviceUrl.Port())
```

Comment:
Was looking at this, and we may need to have a check if `deviceUrl.Port()` is not empty; I did a quick check, and it looks like `net.JoinHostPort` unconditionally adds a colon (`:`), also if there's no port; https://go.dev/play/p/MrZXfaigVRF

```go
package main

import (
	"fmt"
	"net"
)

func main() {
	fmt.Println(net.JoinHostPort("1.2.3.4", ""))
	fmt.Println(net.JoinHostPort("::1", ""))
}
```

Code above prints;

```
1.2.3.4:
[::1]:
```